AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt

```diff
@@ -105,7 +105,7 @@ class SharedDecksDownloadFragment : Fragment() {
         mTryAgainButton = view.findViewById(R.id.try_again_deck_download)
         mCheckNetworkInfoText = view.findViewById(R.id.check_network_info_text)
 
-        val fileToBeDownloaded = arguments?.getSerializable(DOWNLOAD_FILE) as DownloadFile
+        val fileToBeDownloaded = arguments?.getSerializableCompat<Serializable>(DOWNLOAD_FILE) as DownloadFile
```

Comment:
Out of curiosity, since the Compat methods have all sorts of casting and return type logic, why are we using the Compat function here *then* casting via `as DownloadFile` ?

Should it be 

```suggestion
        val fileToBeDownloaded = arguments?.getSerializableCompat<DownloadFile>(DOWNLOAD_FILE)`
```

...or similar, or am I missing something?

Reply:
that was giving an error at line 111 because it requires `DownloadFile` but doing so gives `DownloadFile?`

Reply:
`val fileToBeDownloaded = arguments?.getSerializableCompat<DownloadFile>(DOWNLOAD_FILE)!!` can  we change that to this?

Reply:
So...if I am understanding correctly, it's entirely possible that the de-serialization fails and does not return something.
And the choice is between casting it directly as non-Null DownloadFile or handling the nullability.
Why not handle the nullability? Timber.w the unexpected error case and I think we have a generic "Something went wrong" error message dialog we can pop up?

Reply:
It is always possible to corrupt a type system with direct casting or ignore real nullability concerns with !!, but doing either inevitably leads to errors when the assumption you made while forcing (via cast or !!) is unexpectedly incorrect

Type systems are there to help, and if we agree the types are correct and the nullability is correct in the type system, it's best to pay attention to that instead of forcefully disregarding it

Reply:
> So...if I am understanding correctly, it's entirely possible that the de-serialization fails and does not return something. And the choice is between casting it directly as non-Null DownloadFile or handling the nullability. Why not handle the nullability? Timber.w the unexpected error case and I think we have a generic "Something went wrong" error message dialog we can pop up?

there is a try again option that pops up I guess because it is given down there.

Reply:
So do we need to change anything here or we are good to go ?

Reply:
If you read the code, the fragment should always be initialised

* with arguments
* which include `DOWNLOAD_FILE`

It is an error otherwise. I'm happy to fail at this point if the contract isn't satisfied.

```suggestion
        val fileToBeDownloaded = requireArguments().getSerializableCompat<DownloadFile>(DOWNLOAD_FILE)!!
```

Reply:
If it's truly always initialized, I'd rather see a !! then a cast then, works for me